tests/rptest/services/action_injector.py

```diff
@@ -60,19 +126,18 @@ def max_affected_nodes_reached(self) -> bool:
 
     def target_node(self) -> Optional[ClusterNode]:
         """
-        Randomly selects the next node to apply the action on. A set of affected
-        nodes is maintained so that we do not apply the action on nodes which were
-        already targeted in previous invocations.
+        Selects the next node to process from a priority queue. Initially all nodes have
+        the same priority. When a node has been processed it gets a much lower priority
+        so that it does not get selected again until all the non-affected nodes have been
+        processed.
```

Comment:
Not sure if this is the same thing but I thought that maybe we can just keep two lists for the current and the next generation. We pop off current and put into next, and if current becomes empty, we swap next for current and shuffle it. Maybe this is simpler.

Reply:
Yes, this is much simpler, thanks for the suggestion. I tried it out with two sets and it seems to work. Made the change now.